### ISSUE TITLE
More resilient select field regarding width

### ISSUE THREAD

**Issue by phbernard** [themesberg/flowbite#360]
**Is your feature request related to a problem? Please describe.**

As the select form field is tagged with `w-full`, it is usually large:

![image](https://user-images.githubusercontent.com/423852/207407072-cef67bf1-01ee-499b-ae7d-7a3da8854730.png)

However, if we remove this class, the select's caret now overlaps the content:

![image](https://user-images.githubusercontent.com/423852/207407287-f2b212f3-694b-493e-91d7-9fae8545e1a1.png)

It would be convenient if we could remove `w-full` and still have a good-looking select.

**Describe the solution you'd like**

After we remove `w-full`, we get something like:

![image](https://user-images.githubusercontent.com/423852/207409064-c0262225-0555-4a00-bcf5-a550dbefdab0.png)

In other words, the select field is large enough so the caret never overlaps the content, whatever the selected option.

**Describe alternatives you've considered**

Workarounds:
- Use a fixed minimum width
- Pad options with a few `&nbsp;` (*not* proud of this one)
